### PR TITLE
Try conditional on secrets

### PR DESCRIPTION
**Pull request by twcrone**
Only execute steps/jobs that require AWS/GE keys if they exist to try to prevent lots of failures on forks that cannot access project secrets.

### CODE REVIEW

File: .github/workflows/GHA-Scala-Instrumentation-Tests.yaml [hunk 1]
```diff
@@ -118,7 +134,13 @@ jobs:
           LANGUAGE: scala
           SCALA_VERSION: scala${{ matrix.scala }}
           TEST_TYPE: instrumentation
-        run: ./gradlew --console=plain --parallel :instrumentation:test -Ptest${{ matrix.java-version }} -PnoInstrumentation -PnoScala -Pscala-${{ matrix.scala }} --continue --continue --scan -Dscan.uploadInBackground=false --build-cache
+        run: ./gradlew --console=plain --parallel :instrumentation:test -Ptest${{ matrix.java-version }} -PnoInstrumentation -PnoScala -Pscala-${{ matrix.scala }} --continue --scan -Dscan.uploadInBackground=false --build-cache
+
+      - name: Run instrumentation tests for Java ${{ matrix.java-version }} and scala ${{ matrix.scala }}
```
Comment (meiao) [inline .github/workflows/GHA-Scala-Instrumentation-Tests.yaml:139]: This one is not specifying it does not have GE

File: .github/workflows/GHA-Unit-Tests.yaml [hunk 2]
```diff
@@ -92,6 +94,14 @@ jobs:
         run: |
           ./gradlew --console=plain --parallel test -x :functional_test:test -x :newrelic-scala-api:test -x :newrelic-scala-cats-api:test -x :newrelic-cats-effect3-api:test -x :newrelic-scala-monix-api:test -x :newrelic-scala-zio-api:test -Ptest${{ matrix.java-version }} -PnoInstrumentation --continue --scan -Dscan.uploadInBackground=false --build-cache
 
+      - name: Run unit tests for each Java version as defined in the matrix no build scan or cache
+        if: ${{ env.GRADLE_KEY == '' }}
+        env:
+          JAVA_HOME: ${{ env.ORG_GRADLE_PROJECT_jdk8 }}
+        run: |
+          echo "*** NOT PUBLISHING BUILD SCANS ***"
+          ./gradlew --console=plain --parallel test -x :functional_test:test -x :newrelic-scala-api:test -x :newrelic-scala-cats-api:test -x :newrelic-cats-effect3-api:test -x :newrelic-scala-zio-api:test -Ptest${{ matrix.java-version }} -PnoInstrumentation --continue
```
Comment (meiao) [inline .github/workflows/GHA-Unit-Tests.yaml:103]: this cmd is not skipping the monix-api-test

**Review by meiao — APPROVED** [review]

File: .github/workflows/GHA-Scala-Instrumentation-Tests.yaml (hunk 1 above)
Comment (twcrone) [reply .github/workflows/GHA-Scala-Instrumentation-Tests.yaml:139]: ok, thanks...hard to get these all "right"

File: .github/workflows/GHA-Unit-Tests.yaml (hunk 2 above)
Comment (twcrone) [reply .github/workflows/GHA-Unit-Tests.yaml:103]: dangit!

Comment (twcrone) [reply .github/workflows/GHA-Unit-Tests.yaml:103]: We REALLY need a better way to include/exclude Scala API/Functional tests.  Explicitly is not good IMO.

**Pull request merged**